pkg/block/fetcher.go

```diff
@@ -574,4 +596,37 @@ func (f *ConsistencyDelayMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta,
 			delete(metas, id)
 		}
 	}
+
+	return nil
+}
+
+// DeletionMarkMetaFilter is a MetaFetcher that filters blocks that are marked for deletion (deleteDelay/2) duration before current time.
+type DeletionMarkMetaFilter struct {
+	logger      log.Logger
+	deleteDelay time.Duration
```

Comment:
Do we care here? It's rather `ignoreDelay` time? Also we should comment here, why it's needed.